scripts/prepare_sector_network.py

```diff
@@ -361,44 +370,99 @@ def add_hydrogen(n, costs):
         h2_links = h2_links.groupby("buses_idx").agg(
             {"bus0": "first", "bus1": "first", "length": "mean", "capacity": "sum"}
         )
-    else:
-        attrs = ["bus0", "bus1", "length"]
-        h2_links = pd.DataFrame(columns=attrs)
-
-        candidates = pd.concat(
-            {
-                "lines": n.lines[attrs],
-                "links": n.links.loc[n.links.carrier == "DC", attrs],
-            }
-        )
-
-        for candidate in candidates.index:
-            buses = [candidates.at[candidate, "bus0"], candidates.at[candidate, "bus1"]]
-            buses.sort()
-            name = f"H2 pipeline {buses[0]} -> {buses[1]}"
-            if name not in h2_links.index:
-                h2_links.at[name, "bus0"] = buses[0]
-                h2_links.at[name, "bus1"] = buses[1]
-                h2_links.at[name, "length"] = candidates.at[candidate, "length"]
 
-    # TODO Add efficiency losses
-    if snakemake.config["sector"]["hydrogen"]["network"]:
         if snakemake.config["sector"]["hydrogen"]["gas_network_repurposing"]:
-            n.madd(
-                "Link",
-                h2_links.index + " repurposed",
-                bus0=h2_links.bus0.values + " H2",
-                bus1=h2_links.bus1.values + " H2",
-                p_min_pu=-1,
-                p_nom_extendable=True,
-                p_nom_max=h2_links.capacity.values
-                * 0.8,  # https://gasforclimate2050.eu/wp-content/uploads/2020/07/2020_European-Hydrogen-Backbone_Report.pdf
-                length=h2_links.length.values,
-                capital_cost=costs.at["H2 (g) pipeline repurposed", "fixed"]
-                * h2_links.length.values,
-                carrier="H2 pipeline repurposed",
-                lifetime=costs.at["H2 (g) pipeline repurposed", "lifetime"],
-            )
+            if (
+                not snakemake.config["sector"]["hydrogen"]["network_routes"]
+                == "greenfield"
+            ):
+                # If NOT greenfield, then the existing pipelines routing is used for new and repurposed piepelines.
+                # Else, existing pipelines routing is used for new piepelines only.
+                n.madd(

```

Comment:
From a styling point of view, here we have a lot of nested if/elif that leads to a superlong function.
It seems that it may be nicer to define auxiliary functions to take care of the cases and make the code cleaner, e.g.
```suggestion
if (cond):
  do_this_1(...)
elif (cond 2):
  do_this_2(...)
...
```

What do you think?

Reply:
Done